flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/DefaultVertexParallelismAndInputInfosDecider.java

```diff
@@ -469,4 +469,16 @@ static DefaultVertexParallelismAndInputInfosDecider from(Configuration configura
                         BatchExecutionOptions
                                 .ADAPTIVE_AUTO_PARALLELISM_DEFAULT_SOURCE_PARALLELISM));
     }
+
+    static int getDefaultMaxParallelism(Configuration configuration) {
+        return configuration
+                .getOptional(BatchExecutionOptions.ADAPTIVE_AUTO_PARALLELISM_MAX_PARALLELISM)
+                .orElse(
+                        configuration
+                                .getOptional(CoreOptions.DEFAULT_PARALLELISM)
```

Comment:
I think it should fall back to  `ExecutionConfig#parallelism` instead of the configuration `CoreOptions.DEFAULT_PARALLELISM`.